File: web/src/game/actions/Step.ts
import { Dir } from '../../game2/constants';
import { Creature } from '../../game2/engine/Creature';
import { Action } from './Action';

export class Step implements Action {
  initX: index;
  initY: index;
  readonly direction: Dir;
  readonly duration: ms;
  readonly id: uint;
  readonly creature: Creature;
  readonly time: tsm;

  constructor(id: int, source: Creature, duration: uint, direction: Dir = source.orientation.moving) {

    this.id        = id;
    this.time      = Date.now();//fixme take it from performance*
    this.creature  = source;
    this.initX     = source.orientation.x;
    this.initY     = source.orientation.y;
    this.direction = direction;

    this.duration = duration;
  }


}

File: server/src/main/kotlin/ballad/server/game/actions/Step.kt
package ballad.server.game.actions

import ballad.server.Tsm
import ballad.server.game.Creature
import ballad.server.game.Direction
import ballad.server.game.Duration

data class Step(
    override val x: Int,
    override val y: Int,
    override val time: Tsm,
    val direction: Direction,
    val duration: Duration,
    val creature: Creature,
    var distanceTravelled: Int = 0,
    var finished: Boolean = false
) : Action